feat: Release Chain Permissions

NidhiKJha opened the pull request:
This PR is to remove feature flags and add e2e for Chain Permissions

## **Related issues**

Fixes: [https://github.com/MetaMask/MetaMask-planning/issues/2713](https://github.com/MetaMask/MetaMask-planning/issues/2713)

## **Manual testing steps**

1. Run yarn start
2. Everything should work

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

NidhiKJha committed:
added e2e and removed feature flag

NidhiKJha committed:
reverted permssion page

NidhiKJha committed:
updated tests

NidhiKJha committed:
updated more spec

NidhiKJha committed:
updated more spec

NidhiKJha committed:
updated ui.spec

NidhiKJha committed:
updated prep build and removed unused spec

NidhiKJha committed:
removed unused locale

NidhiKJha committed:
updated spec

NidhiKJha committed:
updated permissions container

NidhiKJha committed:
updated helper

jiexi commented on app/scripts/lib/rpc-method-middleware/handlers/ethereum-chain-utils.js:
```diff
@@ -156,46 +155,34 @@ export function validateAddEthereumChainParams(params, end) {
+      } else {
+        await requestPermittedChainsPermission([chainId]);
       }
-    } else {
-      await requestUserApproval({
-        origin,
-        type: ApprovalType.SwitchEthereumChain,
-        requestData,
-      });
```
@adonesky1 we don't need to keep this around for snaps, right?

adonesky1 replied:
Nope. Snaps don't call `wallet_switchEthereumChain`. And once they do we will be granting them permissions on each request

jiexi commented on app/scripts/lib/rpc-method-middleware/handlers/switch-ethereum-chain.test.js:
```diff
@@ -33,7 +33,7 @@ const createMockLineaMainnetConfiguration = () => ({
   const makeMocks = ({
     permissionedChainIds = [],
-    permissionsFeatureFlagIsActive = false,
+    permissionsFeatureFlagIsActive = true,
```
i believe this can be removed now?

NidhiKJha replied:
Removed!

jiexi commented on app/scripts/metamask-controller.js:
```diff
@@ -5789,8 +5788,7 @@ export default class MetamaskController extends EventEmitter {
         },
-        getChainPermissionsFeatureFlag: () =>
-          Boolean(process.env.CHAIN_PERMISSIONS),
+        getChainPermissionsFeatureFlag: () => true,
```
Think we can just get rid of this entirely?

NidhiKJha replied:
We need it for mocks, can we do that?

jiexi replied:
Sorry I'm not understanding. If this always return true now, then we can just remove all references to `getChainPermissionsFeatureFlag` including the usage in the switch/add ethereum chain handlers and respective tests? Or am I missing something?

NidhiKJha replied:
Ah yes! My bad. We can remove it then

jiexi commented:
I see a few more references to `CHAIN_PERMISSIONS` left when I search in code

NidhiKJha commented:
> I see a few more references to `CHAIN_PERMISSIONS` left when I search in code

Those are for confirmation page Copy Changes

NidhiKJha committed:
updated helper

NidhiKJha committed:
remove getChainPermissionsFeatureFlag variable

NidhiKJha committed:
Merge branch 'develop' into multichain-e2e-tests-v2

jiexi committed:
Fix add/switch chain

NidhiKJha committed:
updated permission page spec

jiexi committed:
Merge remote-tracking branch 'origin/multichain-e2e-tests-v2' into multichain-e2e-tests-v2

# Conflicts:
#	app/scripts/lib/rpc-method-middleware/handlers/switch-ethereum-chain.js

jiexi committed:
fix add/switch

jiexi committed:
Merge branch 'develop' into multichain-e2e-tests-v2

jiexi committed:
Merge remote-tracking branch 'origin/multichain-e2e-tests-v2' into multichain-e2e-tests-v2

adonesky1 commented on app/scripts/metamask-controller.js:
```diff
@@ -5750,10 +5750,9 @@ export default class MetamaskController extends EventEmitter {
-                requestedPermissions[RestrictedMethods.eth_accounts] && {
-                  [PermissionNames.permittedChains]: {},
-                }),
+              ...(requestedPermissions[RestrictedMethods.eth_accounts] && {
+                [PermissionNames.permittedChains]: {},
+              }),
```
> do we anticipate a user trying to request permittedChains by itself? if so we probably need to populate an empty eth_accounts object here too? @adonesky1

🤔 I think whether or not we encourage it we should include an accounts permission as part of the request to avoid confusion/frustration

jiexi committed:
ensure eth_accounts and permittedChains are always incldued together in requestPermissionsForOrigin

NidhiKJha committed:
updated batch txs test

NidhiKJha committed:
updated button

jiexi committed:
Show ChooseAccount on snap account permission request

jiexi committed:
Merge branch 'develop' into multichain-e2e-tests-v2

jiexi committed:
Merge remote-tracking branch 'origin/multichain-e2e-tests-v2' into multichain-e2e-tests-v2

NidhiKJha committed:
chain id spec updated

jiexi committed:
Fix txinsight e2e spec connection

jiexi committed:
Merge remote-tracking branch 'origin/multichain-e2e-tests-v2' into multichain-e2e-tests-v2

NidhiKJha committed:
updated permission Value

NidhiKJha committed:
added review permissions spec file

jiexi committed:
fix switch chain modal assumption

jiexi committed:
Fix request queue ui setup

jiexi committed:
Merge remote-tracking branch 'origin/multichain-e2e-tests-v2' into multichain-e2e-tests-v2

jiexi committed:
Merge branch 'develop' into multichain-e2e-tests-v2

jiexi committed:
Fix dapp-interactions permission spec

jiexi committed:
fix dapp1 switch dapp 2 send!

jiexi committed:
Merge branch 'develop' into multichain-e2e-tests-v2

jiexi committed:
fix request queueing switch chain then watchAsset

jiexi committed:
Merge branch 'develop' into multichain-e2e-tests-v2

jiexi committed:
fix switch custom network spec

jiexi committed:
lint

NidhiKJha committed:
updated edit flow specs

NidhiKJha committed:
lint fix

jiexi committed:
Fix switchChain-sendTx

jiexi committed:
Merge remote-tracking branch 'origin/multichain-e2e-tests-v2' into multichain-e2e-tests-v2

jiexi committed:
fix switchEthereumChain spec

jiexi committed:
Fix request queue signedTypedData

jiexi committed:
Fix request queue signedTypedData undo 7777 change

jiexi committed:
Fix batch-txs-per-dapp-same-network.spec.js

NidhiKJha committed:
fix unit test

NidhiKJha committed:
updated permission cell

NidhiKJha committed:
updated controller

NidhiKJha committed:
prep build fix

NidhiKJha committed:
fixed e2e for test/e2e/tests/network/add-custom-network.spec.js

NidhiKJha committed:
fixed snaps sign e2e

NidhiKJha committed:
remove switch network after add network

NidhiKJha committed:
updated switch ethereum

NidhiKJha committed:
fix dapp interaction spe

NidhiKJha committed:
updated switch ethereum spec

NidhiKJha committed:
fixed test/e2e/accounts/snap-account-signatures-and-disconnects.spec.ts

NidhiKJha committed:
fixed e2e for /home/circleci/project/test/e2e/tests/request-queuing/multiple-networks-dapps-txs.spec.js

NidhiKJha committed:
fixed test/e2e/tests/request-queuing/dapp1-subscribe-network-switch.spec.js

NidhiKJha committed:
fixed e2e for test/e2e/tests/request-queuing/batch-txs-per-dapp-extra-tx.spec.js

NidhiKJha committed:
fixed e2e for test/e2e/tests/request-queuing/batch-txs-per-dapp-diff-network.spec.js

jiexi committed:
Merge branch 'develop' into multichain-e2e-tests-v2

NidhiKJha committed:
removed old unit test for permission connect and updated connection list test

NidhiKJha committed:
removed snapshot

jiexi committed:
Fix api-specs

jiexi committed:
Merge remote-tracking branch 'origin/multichain-e2e-tests-v2' into multichain-e2e-tests-v2

adonesky1 committed:
fix lint

NidhiKJha committed:
updated e2e for test/e2e/tests/request-queuing/multi-dapp-sendTx-revokePermission.spec.js

jiexi committed:
Fix switchEthereumChain "queues switchEthereumChain request from second dapp after send tx request"

jiexi committed:
Merge remote-tracking branch 'origin/multichain-e2e-tests-v2' into multichain-e2e-tests-v2

NidhiKJha committed:
lint fix

NidhiKJha committed:
fixed e2e for test/e2e/json-rpc/wallet_requestPermissions.spec.js

Gudahtt commented on test/e2e/json-rpc/switchEthereumChain.spec.js:
```diff
@@ -268,15 +310,16 @@ describe('Switch Ethereum Chain for two dapps', function () {
-          span: 'span',
-          text: 'Switching networks will cancel all pending confirmations',
-        });
+        // THIS IS BROKEN
```
👀

Gudahtt replied:
It does seem critical for us to continue clearing pending confirmations upon switching networks. The confirmation would no longer be valid after switching. I am seeing at least one flaky test error that looks to be caused by this (here: https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/103391/workflows/9cda560b-8258-4d53-95c9-8f718b4f5473/jobs/3854013/tests)

jiexi replied:
To be clear, switching the network does still clear pending confirmations. There is just no longer any warning to the user a that this will happen. If this was intentional UI/UX, then my label of "THIS IS BROKEN" is incorrect, but otherwise we need to find a way to add it back

jiexi replied:
the test above was broken from flakiness around successive popup/dialogs opening and closing timings

Fixed here https://github.com/MetaMask/metamask-extension/pull/27561/commits/8bf7342e8f6510324bd6322c18a1fc92505efa5b

Gudahtt replied:
I don't think the commit you linked is related to this line (this is a different test)

Gudahtt replied:
Sorry I think I shared the wrong link. I meant to share this one: https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/103391/workflows/7ac38249-dd6e-4a1b-9401-f55c8273fde6/jobs/3853761/tests

Gudahtt replied:
On that failure, see the fourth screenshot, the screenshot with the transaction error on it. It says "Invalid signature v value", which suggests that the chain ID was incorrect.

Either that a pending confirmation was _not_ dismissed upon the network switching, or that there was a race condition where the network switched _after_ a send transaction request was generated, but _before_ it was processed by us.

I'm guessing it's the latter. In which case, extra caution is warranted when performing actions on a dapp just before or after the network has switched. If a confirmation is "in flight" when the network is switched, or if it's triggered immediately after a switch but before the dapp has been told about the new chain ID, we'll see an error like this.

Gudahtt commented on app/scripts/metamask-controller.js:
```diff
@@ -232,6 +232,8 @@ import { getCurrentChainId } from '../../ui/selectors';
 import { getProviderConfig } from '../../ui/ducks/metamask/metamask';
 import { endTrace, trace } from '../../shared/lib/trace';
+// eslint-disable-next-line import/no-restricted-paths
+import { isSnapId } from '../../ui/helpers/utils/snaps';
```
Looks like this should be moved to `/shared`

jiexi committed:
fix batch-txs-per-dapp-extra-tx.spec.js

jiexi committed:
Ensure 8545 is included in connect helper

jiexi committed:
Fix edit-networks-flow to disconnect from 3 to 2 after helpers change

jiexi committed:
Fix api-spec by connect and switch to 1337 without permit dialog

jiexi committed:
Fix test/e2e/accounts/snap-account-signatures-and-disconnects.spec.ts via test/e2e/accounts/common.ts

NidhiKJha committed:
Merge branch 'develop' into multichain-e2e-tests-v2

NidhiKJha committed:
Merge branch 'develop' into multichain-e2e-tests-v2

NidhiKJha committed:
Merge branch 'develop' into multichain-e2e-tests-v2

jiexi committed:
Merge branch 'develop' into multichain-e2e-tests-v2

darkwing committed:
Merge branch 'develop' into multichain-e2e-tests-v2

NidhiKJha committed:
Merge branch 'develop' into multichain-e2e-tests-v2

NidhiKJha committed:
Merge branch 'develop' into multichain-e2e-tests-v2

NidhiKJha committed:
Merge branch 'develop' into multichain-e2e-tests-v2

NidhiKJha committed:
Merge branch 'develop' into multichain-e2e-tests-v2

NidhiKJha committed:
Merge branch 'develop' into multichain-e2e-tests-v2

NidhiKJha committed:
Merge branch 'develop' into multichain-e2e-tests-v2

NidhiKJha committed:
Merge branch 'develop' into multichain-e2e-tests-v2

NidhiKJha committed:
Merge branch 'develop' into multichain-e2e-tests-v2

seaona commented on test/e2e/tests/request-queuing/dapp1-send-dapp2-signTypedData.spec.js:
```diff
@@ -58,24 +57,13 @@ describe('Request Queuing Dapp 1, Switch Tx -> Dapp 2 Send Tx', function () {
         await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
 
         await driver.clickElement({
```
```suggestion
        await driver.clickElementAndWaitForWindowToClose({
```
We should wait until the dialog is closed before opening the new dapp url, otherwise the dapp will never load, see ci failure:

![Screenshot from 2024-10-08 12-14-26](https://github.com/user-attachments/assets/0f495969-8c7b-4f86-b866-9599ff5204b9)

Repro locally too

https://github.com/user-attachments/assets/949d831e-e115-4bd9-ad34-b7ab05ddb0c4

seaona commented on test/e2e/tests/request-queuing/dapp1-send-dapp2-signTypedData.spec.js:
```diff
@@ -58,24 +57,13 @@ describe('Request Queuing Dapp 1, Switch Tx -> Dapp 2 Send Tx', function () {
-        await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
-
-        // Open Dapp Two
+        // Open and connect to Dapp Two
```
```suggestion
        await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);

        // Open and connect to Dapp Two
```
we need to switch back to the test dapp to avoid context is discarded error

NidhiKJha replied:
Updated https://github.com/MetaMask/metamask-extension/pull/27561/commits/a6aee49a9adca593651c1a20072081d1579cbe57

seaona commented on test/e2e/tests/request-queuing/dapp1-send-dapp2-signTypedData.spec.js:
```diff
@@ -110,10 +107,7 @@ describe('Request Queuing Dapp 1, Switch Tx -> Dapp 2 Send Tx', function () {
-        await driver.waitUntilXWindowHandles(4);
-        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
-
-        await driver.clickElement({ text: 'Switch network', tag: 'button' });
```
After that is fixed, we can see a new error: in the test dapp 0, we are switching networks from Mainnet to `0x3e8`,  but when we perform the send from the test dapp 0, the transaction shows Mainnet. I think that's an issue on the wallet level


https://github.com/user-attachments/assets/35fb43d1-5acc-4fee-b73c-9c36b11d4961

NidhiKJha replied:
I tried to replicate it in chrome and it's working there

Gudahtt replied:
@seaona the situation you're describing sounds similar to what I saw here: https://github.com/MetaMask/metamask-extension/pull/27561#discussion_r1787672800

In both cases, the confirmation has the wrong chain

NidhiKJha replied:
Thanks @Gudahtt @seaona , I have pushed this commit that hopefully fix the race condition. https://github.com/MetaMask/metamask-extension/pull/27561/commits/5dbdc39dcb5951b5d53fe8de771cd4b97854695f

NidhiKJha committed:
Merge branch 'develop' into multichain-e2e-tests-v2

NidhiKJha committed:
updated e2e

NidhiKJha committed:
fixed race condition

darkwing committed:
Merge branch 'develop' into multichain-e2e-tests-v2

NidhiKJha committed:
Merge branch 'develop' into multichain-e2e-tests-v2

seaona commented on test/e2e/tests/request-queuing/batch-txs-per-dapp-extra-tx.spec.js:
```diff
@@ -57,34 +57,32 @@ describe('Request Queuing for Multiple Dapps and Txs on different networks', fun
         await switchToNotificationWindow(driver);
 
         await driver.clickElement({
```
```suggestion
        await driver.clickElementAndWaitForWindowToClose({
```

NidhiKJha committed:
Merge branch 'develop' into multichain-e2e-tests-v2

andreahaku reviewed (dismissed):
LGTM

seaona committed:
test: batch tx mmi (#27715)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/27715?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

seaona committed:
test: multi dapp (#27718)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/27718?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

seaona committed:
test: chainid, switch custom net, batch tx diff, snaps (#27725)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/27725?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

NidhiKJha committed:
Merge branch 'develop' into multichain-e2e-tests-v2

seaona commented:
## QA Findings

### Signed Transaction stuck after network switch
I've encountered a case where after switching networks using window.ethereum on a dapp and trying to send a transaction the tx status remains in `Signed`.
 
![Screenshot from 2024-10-09 17-56-07](https://github.com/user-attachments/assets/1b555bed-4a6a-4acc-a1a4-2b8846e75396)

Then I cannot send transactions on other networks, as it says there's a tx pending.

Smart Transactions are disabled.

https://github.com/user-attachments/assets/f7157a65-6d09-4345-8a1c-f994afd5b2b5

I'm in the process of trying to find repro steps

### Invalid v value
This is a failure that sometimes appears in our e2e tests. I think this needs to be investigated further as might be a bug in the wallet side.

![image](https://github.com/user-attachments/assets/5f9b0da8-2dd9-4021-9a4d-4fafcfb9f7da)

I'm trying to find repro steps (no luck so far), but I find it deterministically by running this spec in the mmi build:

`Switch Ethereum Chain for two dapps queues switchEthereumChain request from second dapp after send tx request`

I've seen this in other tests too, but the tests didn't fail as they were not checking if the tx was successful or not. If that's confirmed, we'd need to add a new step to capture this deterministically

seaona committed:
test: should switch to the dapps network automatically when handling sendTransaction calls (#27734)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/27734?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

seaona commented:
## ci e2e webpack: Chrome errors in webpack makes test fail
We see there are some empty errors that make test fails sometimes in webpack. After checking why those are empty we've seen that it's because those RPC errors don't have any value, but a description field. With the below change we can see how the error message is then displayed and we can see how those are expected RPC errors due to the server being down (in the test we shut down the 2 ganache servers)

```
#getErrorFromEvent(event) {
    // Extract the values from the array
    const values = event.args.map((a) => {
      console.log("Argument:", a); // Added 
      return a.value ? a.value : a.description; // changed
  });
```

Current ci:

![image](https://github.com/user-attachments/assets/7b53b0e6-3dd6-446c-9771-735863a6d82b)

With logs added:

![Screenshot from 2024-10-09 20-46-46](https://github.com/user-attachments/assets/ee90aeb3-5854-4469-b18a-eab56d5295ae)

Remaining part: we should fix the `ignore-all` to make that these errors are also ignored.

EtherWizard33 reviewed (commented):
Hey Nidhi, looks like one of the checks in not passing

![2024-10-09 13 17 12](https://github.com/user-attachments/assets/8ddace7e-156b-4d91-b419-e27aa94553fb)

vinnyhoward commented on test/e2e/tests/request-queuing/switchChain-sendTx.spec.js:
```diff
@@ -37,37 +34,54 @@ describe('Request Queuing SwitchChain -> SendTx', function () {
-          text: 'Switching networks will cancel all pending confirmations',
-          tag: 'span',
-        });
+        // await driver.findElement({
```
We can probably remove this correct?

vinnyhoward reviewed (approved):
LGTM!✅

FrederikBolding commented on test/e2e/snaps/test-snap-txinsights.spec.js:
```diff
@@ -37,22 +36,18 @@ describe('Test Snap TxInsights', function () {
           tag: 'button',
         });
 
-        await driver.waitForSelector({ text: 'OK' });
```
I believe these were added to try and combat flakiness,  I don't think we should be removing them.

cc @bowensanders

NidhiKJha replied:
This is replaced with `await driver.clickElementAndWaitForWindowToClose`. We can keep it but here we have have added an extra condition that the windown close when this button is clicked. In other instances of file, we wanted to make sure that the window is closed once we click the button.

FrederikBolding commented on test/e2e/snaps/test-snap-txinsights-v2.spec.js:
```diff
@@ -140,12 +127,6 @@ describe('Test Snap TxInsights-v2', function () {
         });
-
-        // wait for transaction confirmation
-        await driver.waitForSelector({
```
How come this is removed?

NidhiKJha replied:
Yeah, this one I can add back. Can I do it in a follow up PR?

FrederikBolding commented on ui/components/app/permissions-connect-permission-list/permissions-connect-permission-list.js:
```diff
@@ -18,10 +17,6 @@ import { CaveatTypes } from '../../../../shared/constants/permissions';
  * @returns {JSX.Element} A permission description node.
  */
 function getDescriptionNode(permission, index, accounts) {
-  const permissionValue = permission?.permissionValue?.caveats?.find(
```
This seems to revert https://github.com/MetaMask/metamask-extension/commit/4fa502b0d91606513518e32c790d76c7d65aae94 - is that intentional?

NidhiKJha replied:
Yes, because caveats doesn't exist in permissionValue object

FrederikBolding replied:
Hmm, any idea why this value changed since that fix then? 🤔

FrederikBolding commented on ui/components/app/permission-cell/permission-cell-status.js:
```diff
@@ -49,7 +49,7 @@ export const PermissionCellStatus = ({
   const renderAccountsGroup = () => (
     <>
-      {process.env.CHAIN_PERMISSIONS ? (
+      {networks.length > 0 ? (
```
This confuses me, are we stopping rendering `accounts`? If so, can we remove the code for it?

NidhiKJha replied:
We are just keeping it for Snaps. networks is defined only in case of `permittedChains`.

FrederikBolding replied:
Seems like it is in use for non-Snap connections too. I think this deserves a refactor _outside of this PR_. It feels confusing to nest the `networks` display logic inside the account group logic.

<img width="377" alt="Screenshot 2024-10-10 at 11 49 17" src="https://github.com/user-attachments/assets/fbd9a91f-5285-4b33-8a11-d39e50db4c21">

NidhiKJha replied:
For initial connection screen, we need to show the current active account and all the enabled networks to be there. This is connections screen, it doesn't use permission cell anyway

FrederikBolding reviewed (commented):
Looks like we have double padding on `switchEthereumChain`. Unclear if this is caused by this PR
<img width="373" alt="Screenshot 2024-10-10 at 12 11 30" src="https://github.com/user-attachments/assets/7026eadf-871e-4667-88cb-67e720904b92">

FrederikBolding reviewed (approved):
Not blocking, but I have left some comments that should probably be addressed in a follow up.
